### PR TITLE
LPS-122955 Removes SoyPortlet modules

### PR DESCRIPTION
**Pull request by jbalsas**
Unsurprisingly, this removes `SoyPortlet` from DXP, which means that building a soy-based portlet is no longer possible.

We only ever used this inside DXP to build the `ImageEditor` app which was removed recently anyways. This is a step towards [Remove Soy Support from DXP](https://issues.liferay.com/browse/LPS-122954)	in line with our [The State of Frontend Infrastructure](https://liferay.dev/blogs/-/blogs/the-state-of-frontend-infrastructure) announcement 10 months ago.

**Test Plan**
- All relevant tests should be green since this is removing already unused modules
- Verify that a proper BREAKING_CHANGES entry is added announcing that `SoyPortlet` is no longer available

/cc @dsanz, @david-truong

### CODE REVIEW

**Comment by jbalsas** [comment]
ci:test:sf

**Comment by jbalsas** [comment]
ci:test:relevant

**Comment by jbalsas** [comment]
ci:test:sf

**Comment by jbalsas** [comment]
ci:test:relevant

**Comment by jbalsas** [comment]
ci:test:sf

**Comment by jbalsas** [comment]
ci:test:relevant

File: modules/yarn.lock [hunk 1]
```diff
@@ -13704,7 +13704,7 @@ send@0.17.1:
     range-parser "~1.2.1"
     statuses "~1.5.0"
 
-senna@2.7.9, senna@^2.3.1:
+senna@^2.3.1:
```
Comment (wincent) [inline modules/yarn.lock:13707]: 2.3.1, eh. Nice.

File: readme/BREAKING_CHANGES.markdown [hunk 2]
```diff
@@ -340,4 +340,45 @@ By moving minification of frontend resources from run-time to build-time
...
+The class `SoyPortlet` used to implement Portlet whose views are backed by
+Closure Templates (Soy) has been removed and is no longer available.
+
+#### Who is affected?
+
+Anyone using `SoyPortlet` as a base for their portlet developments.
+
+#### How should I update my code?
+
+We heavily recommend re-writing your Soy portlets using either a well stablished
```
Comment (wincent) [inline readme/BREAKING_CHANGES.markdown:360]: stablished → established

File: readme/BREAKING_CHANGES.markdown [hunk 3]
```diff
@@ -340,4 +340,45 @@ By moving minification of frontend resources from run-time to build-time
...
+architecture such as MCVPortlet using JSPs or a particular frontend framework of
```
Comment (wincent) [inline readme/BREAKING_CHANGES.markdown:361]: Should `MCVPortlet` be in backticks? Also, should it be **MVC**Portlet?

File: readme/BREAKING_CHANGES.markdown [hunk 4]
```diff
@@ -340,4 +340,45 @@ By moving minification of frontend resources from run-time to build-time
...
+your choice.
+
+As a temporary measure, you could alternatively copy all the necessary removed
+classes into our own. However, support for soy templates is likely to be removed
```
Comment (wincent) [inline readme/BREAKING_CHANGES.markdown:365]: our → your

soy → Soy

File: readme/BREAKING_CHANGES.markdown [hunk 5]
```diff
@@ -340,4 +340,45 @@ By moving minification of frontend resources from run-time to build-time
...
+in this version as well so doing this might require a lot of work.
+
+#### Why was this change made?
+
+This is done as a way to simplify our frontend technical offering and better
+focus on proven technologies with high demand on the market.
```
Comment (wincent) [inline readme/BREAKING_CHANGES.markdown:371]: on the market → in the market

File: readme/BREAKING_CHANGES.markdown [hunk 6]
```diff
@@ -340,4 +340,45 @@ By moving minification of frontend resources from run-time to build-time
...
+
+A further exploration and analysis of the different frontend options available
+can be found in [The State of Frontend Infrastructure](https://liferay.dev/blogs/-/blogs/the-state-of-frontend-infrastructure) including a rationale on why we're moving
+away from soy:
+
+> Liferay has invested several years into Soy believing it was the holy grail.
```
Comment (wincent) [inline readme/BREAKING_CHANGES.markdown:377]: Hmmm... a religious reference!

File: readme/BREAKING_CHANGES.markdown [hunk 7]
```diff
@@ -340,4 +340,45 @@ By moving minification of frontend resources from run-time to build-time
...
+> We believed the ability to compile closure templates would provide us the
```
Comment (wincent) [inline readme/BREAKING_CHANGES.markdown:378]: closure → Closure

**Review by wincent — COMMENTED** [review]
Screenshot 2020-12-10 -114859-9AFUPu5J@2x

in-love

Although there are some copy edits to make in the BREAKING_CHANGES.markdown.

File: modules/yarn.lock (hunk 1 above)
Comment (jbalsas) [reply modules/yarn.lock:13707]: Huh... we almost managed to get rid of `senna` but still bringing one over through `metal-router` in `frontend-js-metal-web`... so close... :D

File: readme/BREAKING_CHANGES.markdown (hunk 6 above)
Comment (jbalsas) [reply readme/BREAKING_CHANGES.markdown:377]: 🙏

File: readme/BREAKING_CHANGES.markdown (hunk 2 above)
Comment (jbalsas) [reply readme/BREAKING_CHANGES.markdown:360]: > stablish [ stab-lish ]
> 
> verb (used with object) Archaic

I'm archaic 😂

File: readme/BREAKING_CHANGES.markdown (hunk 3 above)
Comment (jbalsas) [reply readme/BREAKING_CHANGES.markdown:361]: Nice catch!

**Comment by jbalsas** [comment]
Hey @wincent, I amended the `BREAKING_CHANGES` with your suggestions and added a couple more removals while I was at it:

Screen Shot 2020-12-10 at 11 21 32

Let's see if this is still green later 😂

**Comment by jbalsas** [comment]
ci:test:sf

**Comment by jbalsas** [comment]
Hey @liferay-lima, just cc'ing you here since I think you'll be okay with:
- [LPS-122964 Removes unused Soy Flags Tag](https://github.com/liferay-frontend/liferay-portal/pull/602/commits/8d2dceb7b105bb886f51fcc1e008937f34ee4c47)
- [sharing-notifications/build.gradle](https://github.com/liferay-frontend/liferay-portal/pull/602/files#diff-cc8140bcf6c24055546276915b312809b25002abd170b8e30cc1307fb41ce632) and [sharing-web/build.gradle](https://github.com/liferay-frontend/liferay-portal/pull/602/files#diff-211c3cb08b7f55bc8b474d434dc6c4b69b1541ee3580227758ebcf92311fbdc6)

**Comment by jbalsas** [comment]
ci:test:relevant

**Comment by jbalsas** [comment]
@wincent, do you have final words about this? I'm going to forward it manually if I have the chance :D

**Comment by jbalsas** [comment]
@john-co, @PablitoBonito, @manoelcyreno, just notice how we got almost the same failure both as unique and in common... asked about this in https://liferay.slack.com/archives/C19PWQ74J/p1607614522247700 in case you're curious

**Comment by jbalsas** [comment]
ci:forward

**Comment by manoelcyreno** [comment]
ci:test:relevant

**Comment by austinchiang** [comment]
ci:test:relevant

**Comment by jbalsas** [comment]
ci:test:sf

**Comment by jbalsas** [comment]
ci:test:relevant

**Comment by jbalsas** [comment]
Off we go!

**Comment by jbalsas** [comment]
ci:forward

**Pull request closed without merging**